invenio_oaiserver/models.py

```diff
@@ -77,4 +80,20 @@ class OAISet(db.Model, Timestamp):
     """Search pattern to get records."""
 
 
+def oaiset_removed_or_inserted(mapper, connection, target):
+    """Invalidate cache on collection insert or delete."""
+    current_oaiserver.sets = None
+
+
+def oaiset_attribute_changed(target, value, oldvalue, initiator):
+    """Invalidate cache if dbquery change."""
+    if value != oldvalue:
+        current_oaiserver.sets = None
+
+
+# update cache with list of collections
+listen(OAISet, 'after_insert', oaiset_removed_or_inserted)
+listen(OAISet, 'after_delete', oaiset_removed_or_inserted)
+listen(OAISet.search_pattern, 'set', oaiset_attribute_changed)
```

Comment:
I think the `set` event is called every time you do something like `myset.search_pattern = 'fuu bar'`.
If this is the behaviour, I'm not sure if it's what we need because you are not sure that the changes are commited.